src/Parser/ArrayMapArgVisitor.php

```diff
@@ -14,19 +14,41 @@ final class ArrayMapArgVisitor extends NodeVisitorAbstract
 
 	public function enterNode(Node $node): ?Node
 	{
-		if ($node instanceof Node\Expr\FuncCall && $node->name instanceof Node\Name && !$node->isFirstClassCallable()) {
-			$functionName = $node->name->toLowerString();
-			if ($functionName === 'array_map') {
-				$args = $node->getArgs();
-				if (isset($args[0])) {
-					$slicedArgs = array_slice($args, 1);
-					if (count($slicedArgs) > 0) {
-						$args[0]->value->setAttribute(self::ATTRIBUTE_NAME, $slicedArgs);
-					}
-				}
-			}
+		if (!$this->isArrayMapCall($node)) {
+			return null;
 		}
+
+		$args = $node->getArgs();
+		if (count($args) < 2) {
+			return null;
+		}
+
+		$callbackPos = 0;
+		if ($args[1]->name !== null && $args[1]->name->name === 'callback') {
+			$callbackPos = 1;
+		}
+		[$callback] = array_splice($args, $callbackPos, 1);
```

Comment:
I have no idea what's going on at this line. Please make the code easier to read.

Reply:
I made some refactors.